File: src/main/java/org/BeehiveRobotics/Library/Motors/MotorModel.java
package org.BeehiveRobotics.Library.Motors;

public enum MotorModel {
    NEVEREST20, NEVEREST40, NEVEREST60;
    public double CPR;
    public static double DEFAULT_CPR = 0;
    public static double CPR(MotorModel motorModel) {
        switch (motorModel){
            case NEVEREST20:
                return 537.6;
            case NEVEREST40:
                return 1120;
            case NEVEREST60:
                return 1680;
        }
        return DEFAULT_CPR;
    }
    public int RPM;
    public static int DEFAULT_RPM = 0;
    public static int RPM(MotorModel motorModel) {
        switch (motorModel) {
            case NEVEREST20:
                return 340;
            case NEVEREST40:
                return 160;
            case NEVEREST60:
                return 105;
        }
        return DEFAULT_RPM;
    }
}

File: src/main/java/org/BeehiveRobotics/Library/Motors/Kotlin/MotorModel.kt
package org.BeehiveRobotics.Library.Motors.Kotlin

enum class MotorModel {
    NEVEREST20, NEVEREST40, NEVEREST60;

    var CPR: Double = 0.toDouble()
    var DEFAULT_CPR = 0.0
    fun CPR(motorModel: MotorModel): Double {
        when (motorModel) {
            NEVEREST20 -> return 537.6
            NEVEREST40 -> return 1120.0
            NEVEREST60 -> return 1680.0
        }
        return DEFAULT_CPR
    }

    var RPM: Int = 0
    var DEFAULT_RPM = 0
    fun RPM(motorModel: MotorModel): Int {
        when (motorModel) {
            NEVEREST20 -> return 340
            NEVEREST40 -> return 160
            NEVEREST60 -> return 105
        }
        return DEFAULT_RPM
    }

}